test/hotspot/jtreg/serviceability/jvmti/FollowReferences/FieldIndices/libFieldIndicesTest.cpp

```diff
@@ -0,0 +1,517 @@
+        return nullptr;
+    }
+    return klass->fields[index].name;
+}
+
+
+jint JNICALL primitiveFieldCallback(
+     jvmtiHeapReferenceKind        reference_kind,
+     const jvmtiHeapReferenceInfo* reference_info,
+     jlong                         class_tag,
+     jlong*                        tag_ptr,
+     jvalue                        value,
+     jvmtiPrimitiveType            value_type,
+     void*                         user_data)
+{
+    if (*tag_ptr == 0) {
+        return 0;
+    }
+
+    jint index = reference_info->field.index;
+    jint int_value = value.i;
+    if (value_type != JVMTI_PRIMITIVE_TYPE_INT) {
+        printf("ERROR: unexpected value type in primitiveFieldCallback: '%c'\n", (char)value_type);
+        test_failed = true;
+        int_value = -1;
+    }
+
+    if (reference_kind == JVMTI_HEAP_REFERENCE_FIELD) {
+        Object* obj = (Object*)(*tag_ptr);
+        Klass* klass = obj->klass;
+        printf("primitiveFieldCallback(JVMTI_HEAP_REFERENCE_FIELD): "
+               "klass=%s, index=%d, type=%c, value=%d\n",
+               klass->name, index,
+               (int)value_type, (int)value.i);
+        if (check_index_bounds(index, klass)) {
+            jint expected_value = obj->field_values[index - klass->interface_field_count];
+            if (int_value != expected_value) {
+                printf("  ERROR: wrong instance value: (%d, expected %d)\n",
+                       (int)int_value, (int)expected_value);
+                test_failed = true;
+            } else {
+                printf("  OK: field %s.%s, value %d\n",
+                       klass->name, get_field_name(klass, index), (int)int_value);
+            }
+        }
+    } else if (reference_kind == JVMTI_HEAP_REFERENCE_STATIC_FIELD) {
+        Klass* klass = (Klass*)(*tag_ptr);
+        printf("primitiveFieldCallback(JVMTI_HEAP_REFERENCE_STATIC_FIELD): "
+               "klass=%s, index=%d, type=%c, value=%d\n",
+               klass->name, index,
+               (int)value_type, (int)value.i);
+        if (check_index_bounds(index, klass)) {
+            jint expected_value = klass->fields[index - klass->interface_field_count].value;
+            if (int_value != expected_value) {
+                printf("  ERROR: wrong static value: (%d, expected %d)\n\n\n",
+                       (int)int_value, (int)expected_value);
+                test_failed = true;
+            } else {
+                printf("  OK: field %s.%s, value %d\n",
+                          klass->name, get_field_name(klass, index), (int)int_value);
+            }
+        }
+    } else {
+        printf("ERROR: unexpected reference_kind in primitiveFieldCallback: %d\n", (int)reference_kind);
+        test_failed = true;
+    }
+
+    fflush(0);
+    return 0;
+}
+
+
+JNIEXPORT void JNICALL
+Java_FieldIndicesTest_prepare(JNIEnv *env, jclass cls, jobject testObj) {
+    Object::explore(env, testObj);
+    fflush(0);
```

Comment:
stdout?

Reply:
fflush(0) flashes all opened streams (including stout and stderr).
It's to handle the case when some shared test routines log to stderr.
But flash() expects `FILE *`, so I updated all calls to use `nullptr`